src/legacy/core_plugins/kibana/public/context/index.js

```diff
@@ -29,26 +29,34 @@ import { getRootBreadcrumbs } from '../discover/breadcrumbs';
 import { npStart } from 'ui/new_platform';
 import { subscribeWithScope } from 'ui/utils/subscribe_with_scope';
 
+const k7Breadcrumbs = $route => {
+  const { indexPattern } = $route.current.locals;
+  const { id } = $route.current.params;
+
+  return [
+    ...getRootBreadcrumbs(),
+    {
+      text: i18n.translate('kbn.context.breadcrumb', {
+        defaultMessage: 'Context of {indexPatternTitle}#{docId}',
+        values: {
+          indexPatternTitle: indexPattern.title,
+          docId: id,
+        },
+      }),
+    },
+  ];
+};
+
+
 uiRoutes
+  // deprecated route, kept for compatibility
+  // should be removed in the future
   .when('/context/:indexPatternId/:type/:id*', {
+    redirectTo: '/context/:indexPatternId/:id'
+  })
+  .when('/context/:indexPatternId/:id*', {
```

Comment:
note: this is the main change here, other changes are a bit of refactoring + auto formatting